### PR TITLE
Implement SDL 0099 Remote Control - LIGHT, AUDIO, HMI_SETTINGS and parameters (SIS Data)

### LINKED ISSUE #755: [SDL 0099] New remote control modules (LIGHT, AUDIO, HMI_SETTINGS) and parameters (SIS Data)
Proposal: [New remote control modules (LIGHT, AUDIO, HMI_SETTINGS) and parameters (SIS Data)](https://github.com/smartdevicelink/sdl_evolution/blob/master/proposals/0099-new-remote-control-modules-and-parameters.md)

> SDL remote control provides the ability for mobile applications to control certain components of the vehicle, such as, tune the radio, set the desired temperature or change the fan speed. However, currently only `CLIMATE` and `RADIO` modules are available for remote-control. Mobile application developers need more than basic climate or radio control. To meet developers' requirements, we propose to add three new modules (`AUDIO`, `LIGHT`, `HMI_SETTINGS`) and some new parameters (heatedXXXEnable and radio sisData) to the existing `CLIMATE` and `RADIO` modules respectively. This proposal only defines new data types and new parameters added to the existing data types. It does not add or change any remote control RPCs.

Review: https://github.com/smartdevicelink/sdl_evolution/issues/296

Steering Committee Decision:

> The Steering Committee voted to accept this proposal with revisions. The author will revise the proposal to fix a few typos, as well as include descriptions for the groups of lights - listing which single light elements they typically include. It was also noted that limiting the scope of proposals, especially those related to Remote Control, in the future will help to make for easier understanding and discussion among SDLC Members.
>Proposal has now been revised to include the descriptions and fix the typos mentioned.

### PR DESCRIPTION
**Pull request by mrapitis**
Fixes #755 

This PR is **ready** for review.

## Risk

This PR makes **minor** API changes.

## Testing Plan

Added new unit tests and expanded existing ones

## Summary

Added RPC's, structs, enums as defined in the Remote Control -LIGHT, AUDIO, HMI_SETTINGS and parameters (SIS Data) proposal.

## CLA

- [x] I have signed the CLA

### CODE REVIEW

**Comment by joeljfischer** [comment]
@mrapitis Should this be closed in favor of #1038?

**Comment by mrapitis** [comment]
@joeljfischer both #1038 and #1036 have this PR as a dependency.  If desired, we could potentially combine the three PR's into a single PR if this simplifies things without causing confusion?

**Comment by joeljfischer** [comment]
@mrapitis 

TL;DR, yes, please.

They're not quite a dependency since they each include the previous PR as their own commit (as opposed to targetting the branch of the dependent code). However, I didn't notice that #1036 also shared this code. One PR containing the proposal and its revision proposals would be ideal.

**Comment by mrapitis** [comment]
Closing in favor of #1038
